programs/add_ptr_stack.zasm

```diff
@@ -0,0 +1,16 @@
+	.text
+	.file	"add.zasm"
+	.globl	__entry
+__entry:
+.func_begin0:
+	; sets r1=FatPointer from test
+	add 5,r0,r2
+	add r0,r0,stack+=[1]
+	ptr.add r1,r0,stack=[0]
```

Comment:
How does ptr.add work? I dont' remember seeing this on the spec.

Reply:
ptr.add receives two inputs, the first one is a Fat pointer, with the following fields:
`{offset, page, start, len }`
The second input is a number usually called diff, and the result of ptr.add is a new pointer `{offset + diff, page, start, len}`
You can find the specification on page 205 on the EraVM ISA Specification